Encryption.md

```diff
@@ -132,49 +203,164 @@ the encryption buffer is comprised of an IV, ciphertext and tag, as described in
 Algorithms section. The length of the encryption buffer (a 4-byte little endian) is 
 written to the output stream, followed by the buffer itself.
 
+|length (4 bytes) | IV (12 bytes) | ciphertext (`length` bytes) | tag (16 bytes) |
+|-----------------|---------------|-----------------------------|----------------|
+
 The column pages (data and dictionary) are encrypted after compression. For each page, 
-the encryption buffer is comprised of an IV, ciphertext and (in case of AES_GCM_V1) of a 
+the encryption buffer is comprised of an IV (nonce), ciphertext and (in case of AES_GCM_V1) of a 
 tag, as described in the Algorithms section. Only the buffer is written to the output 
-stream - not need to write the length of the buffer, since the length (size of the page after
+stream - no need to write the length of the buffer, since the length (page size after
 compression and encryption) is kept in the page headers.
 
-A `crypto_meta_data` field in set in each `ColumnChunk` in the encrypted columns. 
+| IV (12 bytes) | ciphertext (`page_size` bytes) | tag (16 bytes) |
+|---------------|--------------------------------|----------------|
+
+or 
+
+| IV (16 bytes) | ciphertext (`page_size` bytes) | 
+|---------------|--------------------------------|
+
+
+A `crypto_meta_data` field is set in each `ColumnChunk` in the encrypted columns. 
 `ColumnCryptoMetaData` is a union - the actual structure is chosen depending on whether the 
 column is encrypted with the footer key, or with a column-specific key. For the latter, 
 a key metadata can be specified, with a maximal length of 512. Key metadata is a free-form
-byte array that can be used by a reader to retrieve the column encryption key. 
-
-Parquet file footer, and its nested structures, contain sensitive information - ranging 
+byte array that can be used by a reader to retrieve the encryption key (see next subsection). 
```

Comment:
It should be clear exactly which structures are encrypted using the key specified here. This is used for any column-specific encryption, right? So that would be all page headers, page data buffers, the column index, the offset index, and the column metadata?